reana/reana_dev/run.py

```diff
@@ -538,28 +555,46 @@ def run_example(
                             or "stopped" in status
                         ):
                             break
-                    # verify logs message presence
-                    for log_message in get_expected_log_messages_for_example(component):
-                        cmd = "reana-client logs -w {0} | grep -c '{1}'".format(
-                            workflow_name, log_message
+
+                # check only is here
+                if not submit_only:
+                    cmd = f"reana-client status -w {workflow_name}"
+                    status = run_command(cmd, component, return_output=True)
+
+                    if "running" in status:
+                        run_statistics["running"].append(workflow_name)
+
+                    if "failed" in status:
+                        run_statistics["failed"].append(workflow_name)
+                        continue
+
+                    if "finished" in status or "stopped" in status:
+
+                        if not _verify_log_output(component, workflow_name):
+                            run_statistics["failed"].append(workflow_name)
+                            continue
+
+                        missing_files = _return_missing_output_files(
+                            component, workflow_name
                         )
-                        run_command(cmd, component)
-                    # verify output file presence
-                    cmd = "reana-client ls -w {0}".format(workflow_name)
-                    listing = run_command(cmd, component, return_output=True)
-                    click.secho(listing)
-                    expected_files = file or get_expected_output_filenames_for_example(
-                        component
-                    )
-                    for expected_file in expected_files:
-                        if expected_file not in listing:
-                            click.secho(
-                                "[ERROR] Expected output file {0} not found. "
-                                "Exiting.".format(expected_file)
-                            )
-                            sys.exit(1)
-    # report that everything was OK
-    run_command("echo OK", component)
```

Comment:
It is still good to say final OK or FAILED, just to have fully explicit confirmation about the final outcome of tests. (And to keep continuity with what we had so far.) Also, please don't forget to exit 0 when OK, exit 2 when FAILED, and exit 1 when in 'PROGRESS' (e.g. when some workflows are still running and cannot be analysed).

Reply:
Upon seeing `PROGRESS`, it may not be fully clear to people, we would have to say `IN_PROGRESS` perhaps... I'd propose to simply say `RUNNING` which will create a nice reference to "2 running".  So we'd have 0=OK, 1=RUNNING, 2=FAILED statuses.  WDYT?

Reply:
I used "RUNNING".